docs/how-to/nat-configuration.md

```diff
@@ -1,32 +1,170 @@
+A common issue for new users running IPFS Kubo, especially on home networks, is that their IPFS node is behind a [NAT (network address translation)](../concepts/glossary.md#nat) layer (such as a residential network router) experience very long wait times or high rate of request failure.
+Connection issues that arise as a result of a NAT can be solved with various router configurations, such as port forwarding, each of which is discussed in this guide. 
+
+
+## Background 
+
+IPFS nodes behind a NAT often have difficulty connecting to the rest of the nodes on the IPFS network. However, there are many benefits to a NAT, because it:
+
+- Allows many machines to share a single public address
+- Is essential for the continued functioning of the IPv4 protocol, which would otherwise be unable to serve the needs of the modern networked population with its 32-bit address space.
 
 When you connect to your home WiFi, your computer gets an IPv4 address of something like `10.0.1.15`. This is part of a range of IP addresses reserved for internal use by private networks. When you make an outgoing connection to a public IP address, the router replaces your internal IP with its own public IP address. When data comes back from the other side, the router will translate back to the internal address.
 
-While the NAT is usually transparent for outgoing connections, listening for incoming connections requires some configuration. The router listens on a single public IP address, but any number of machines on the internal network could handle the request. To serve requests, your router must be configured to send certain traffic to a specific machine. You can configure your router to allow these connections by either of the following sections.
+While NATs are generally transparent for outgoing connections, listening for incoming connections requires some configuration. The router listens on a single public IP address, but any number of machines on the internal network could handle the request. To serve requests, your router must be configured to send certain traffic to a specific machine. 
+
+## Configuration Options
 
-## IPv6
+The appropriate configuration option for your router depends on your specific setup:
+
+- If your router supports them, [enable IPv6](#enable-ipv6) or [enable UPnP](#enable-upnp) to solve most connection issues
+- [Disable DCUtR holepunching](#disable-dcutr-holepunching), which is enabled by default as of Kubo v0.13
+- If IPv6 or UPnP are not available, [enable manual port forwarding](#enable-manual-port-forwarding)
+
+### Enable IPv6
 
 If your router and internet service provider (ISP) support IPv6, enabling it will mitigate some connection issues. Most modern routers have a single option to enable IPv6. We are unable to give detailed information about each router's settings and preferences here. Search your router manufacturer's website for _IPv6_ for information on how to enable it.
```

Comment:
Add a disclaimer here IPv6 does not always solve everything, my isp have per machine ipv6 but it still run a firewall that blocks inbound connections.
I belive that is fairly commun.